cloudinit/sources/tests/test_init.py

```diff
@@ -336,7 +336,7 @@ def test_get_data_writes_json_instance_data_sensitive(self):
         sensitive_json_file = self.tmp_path(INSTANCE_JSON_SENSITIVE_FILE, tmp)
         redacted = util.load_json(util.load_file(json_file))
         self.assertEqual(
-            {'cred1': 'sekret', 'cred2': 'othersekret'},
+            REDACT_SENSITIVE_VALUE,
```

Comment:
Looking at the unittest, I think we should refactor it.  Currently, 

def test_get_data_writes_json_instance_data_sensitive(self):               
        """get_data writes INSTANCE_JSON_SENSITIVE_FILE as readonly root."""

has this stanza:

```
        json_file = self.tmp_path(INSTANCE_JSON_FILE, tmp)                     
        sensitive_json_file = self.tmp_path(INSTANCE_JSON_SENSITIVE_FILE, tmp) 
        redacted = util.load_json(util.load_file(json_file))                   
        self.assertEqual(                                                      
            {'cred1': 'sekret', 'cred2': 'othersekret'},                       
            redacted['ds']['meta_data']['some']['security-credentials'])  
```

I think this unittest is doing more than it needs to given the doc-string.

We should call get_data(), and then load the *sensitive* file only:
1) test that the file is written, and the filename matches INSTANCE_JSON_SENSITIVE_FILE
2) test the permissions
3) assert that the datasource.sensitive_metadata_keys are in the expected_result dict
4) assert that the value of expected_result[SENSITIVE_KEY] matches the original value.

And lastly, this unittest should not include the word REDACTED in any of the variables at all as
when we're handling the sensitive file, there is not redaction at all.

And a second test should cover the redacted scenario, mirroring the above, and should not contain
any variables or comments mentioning 'sensitive'.